source/agora/node/Ledger.d

```diff
@@ -1346,3 +1352,50 @@ unittest
     genBlockTransactionsFreeze(1, TxType.Payment);
     assert(ledger.getBlockHeight() == 11 + 2016);
 }
+
+/// test that loading wrong genesis block throws
+unittest
+{
+    import agora.common.crypto.Key;
+    import agora.common.Types;
+    import agora.common.Hash;
+    import agora.common.Serializer;
+    import std.conv;
+
+    auto gen_key_pair = getGenesisKeyPair();
+
+    immutable gen_block_hex = "0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000055AF5DACF2021214C9F691408C2B9F5A2A655F6660D1C9A582641D005B20B90261E508ED0611EF59A52944F6E677D101F416D29F931138D313BCB94074B507800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008FF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FDFF001013E5A2CC000061125F97FA2B27586CA8851DDA9E9BAF369E06B61874D26E23145A2D425705FD01055AF5DACF2021214C9F691408C2B9F5A2A655F6660D1C9A582641D005B20B90261E508ED0611EF59A52944F6E677D101F416D29F931138D313BCB94074B5078";
+
+    // hex => bin
+    auto block_bytes = gen_block_hex.chunks(2).map!(
+        twoDigits => twoDigits.parse!ubyte(16)).array();
+    immutable new_gen_block = block_bytes.deserializeFull!(immutable(Block));
+    auto storage = new MemBlockStorage();
+    storage.saveBlock(new_gen_block);  // stores it as the genesis block
+
+    auto pool = new TransactionPool(":memory:");
+    scope(exit) pool.shutdown();
+    auto utxo_set = new UTXOSet(":memory:");
+    scope (exit) utxo_set.shutdown();
+    auto config = new Config();
+    auto enroll_man = new EnrollmentManager(":memory:", gen_key_pair);
+    scope (exit) enroll_man.shutdown();
+    config.node.is_validator = true;
+
+    try
+    {
+        scope ledger = new Ledger(pool, utxo_set, storage, enroll_man, config.node);
```

Comment:
Can you replace this with a `TestLedger` ?

Reply:
I don't think so, I need to pass the MemBlockStorage loaded with the wrong block to the constructor of Ledger.

Reply:
`TestLedger` accepts an array of `Block` as second argument which will then be loaded by the `MemBlockStorage`.